stories/playlist.tsx

```diff
@@ -1,47 +1,85 @@
 import React, { Component } from 'react'
 import AudioPlayer from '../src/index'
 
-const playlist = [
-  { name: 'SoundHelix-Song-9', src: 'https://www.soundhelix.com/examples/mp3/SoundHelix-Song-9.mp3' },
-  { name: 'SoundHelix-Song-4', src: 'https://www.soundhelix.com/examples/mp3/SoundHelix-Song-4.mp3' },
-  { name: 'SoundHelix-Song-8', src: 'https://www.soundhelix.com/examples/mp3/SoundHelix-Song-8.mp3' },
-]
+interface Track {
+  src: string
+  type: string
+  additionalSrcs?: [
+    {
+      src: string
+      type: string
+    }
+  ]
+}
+
+interface PlaylistProps {
+  playlist: Track[]
+  useSourceElements?: boolean
+}
 
 interface PlayListState {
   currentMusicIndex: number
 }
 
-class PlayList extends Component<null, PlayListState> {
+class PlayList extends Component<PlaylistProps, PlayListState> {
+  playlist: Track[]
+  useSourceElements?: boolean = false
+
   state = {
     currentMusicIndex: 0,
   }
 
+  constructor(props: PlaylistProps) {
+    super(props)
+    this.playlist = props.playlist
+    this.useSourceElements = props.useSourceElements
+  }
+
   handleClickPrevious = (): void => {
     this.setState((prevState) => ({
-      currentMusicIndex: prevState.currentMusicIndex === 0 ? playlist.length - 1 : prevState.currentMusicIndex - 1,
+      currentMusicIndex: prevState.currentMusicIndex === 0 ? this.playlist.length - 1 : prevState.currentMusicIndex - 1,
     }))
   }
 
   handleClickNext = (): void => {
     this.setState((prevState) => ({
-      currentMusicIndex: prevState.currentMusicIndex < playlist.length - 1 ? prevState.currentMusicIndex + 1 : 0,
+      currentMusicIndex: prevState.currentMusicIndex < this.playlist.length - 1 ? prevState.currentMusicIndex + 1 : 0,
     }))
   }
 
   render(): React.ReactNode {
     const { currentMusicIndex } = this.state
+    const track = this.playlist[currentMusicIndex]
+
     return (
       <div>
         <p>currentMusicIndex: {currentMusicIndex}</p>
-        <AudioPlayer
-          onEnded={this.handleClickNext}
-          autoPlayAfterSrcChange={true}
-          showSkipControls={true}
-          showJumpControls={false}
-          src={playlist[currentMusicIndex].src}
-          onClickPrevious={this.handleClickPrevious}
-          onClickNext={this.handleClickNext}
-        />
+
+        {this.useSourceElements ? (
+          <AudioPlayer
+            srcKey={track.src}
+            onEnded={this.handleClickNext}
+            autoPlayAfterSrcChange={true}
+            showSkipControls={true}
+            showJumpControls={false}
+            onClickPrevious={this.handleClickPrevious}
+            onClickNext={this.handleClickNext}
+          >
+            <source src={track.src} type={track.type} />
+            {track.additionalSrcs &&
+              track.additionalSrcs.map(({ src, type }) => <source key={src} src={src} type={type} />)}
+          </AudioPlayer>
```

Comment:
I feel using `srcKey` is confusing since in a native audio tag, it either has `src` attribute or `source` children. could you try an approach that only uses `source`s without adding a new prop?

Reply:
@lhz516 what do you think of the changes?